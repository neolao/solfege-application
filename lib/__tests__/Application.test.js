const Application = require("../Application");

describe("Application", () => {
  let application;
  beforeEach(() => {
    application = new Application;
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("start()", () => {
    it("should emit a start event", async () => {
      const spy = jest.spyOn(application, "emit");
      await application.start();

      expect(spy).toBeCalledWith(Application.EVENT_START, application, []);
    });

    it("should emit an end event at the end", async () => {
      const spy = jest.spyOn(application, "emit");
      await application.start();

      expect(spy).toHaveBeenLastCalledWith(Application.EVENT_END, application);
    });
  });

  describe("setParameter()", () => {
    it("should store parameter", () => {
      application.setParameter("foo", "bar");

      expect(application.getParameter("foo")).toEqual("bar");
    });

    it("should throw an error if the parameter name is not a string", () => {
      expect(() => {
        application.setParameter(42, "bar");
      }).toThrowErrorMatchingSnapshot();
    });
  });

  describe("addBundle()", () => {
    it("should store bundle", () => {

    });
  });
});
