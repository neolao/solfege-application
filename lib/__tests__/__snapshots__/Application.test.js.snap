// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application addBundle() should throw an error if the bundle does not have getPath() method 1`] = `"TypeError: Invalid bundle, getPath() method is missing"`;

exports[`Application addBundle() should throw an error if the bundle.getPath() does not return a directory 1`] = `"Invalid bundle, getPath() does not return a directory"`;

exports[`Application setParameter() should throw an error if the parameter name is not a string 1`] = `"TypeError: Parameter name should be a string, invalid type: number"`;
